### PR TITLE
Add private key to pkcs8 passphrase

### PR DESCRIPTION
**Pull request by sven-hm**
This PR adds the counterparts of `d2i_PKCS8PrivateKey_bio` and `private_key_from_pkcs8_passphrase`.

### CODE REVIEW

**Comment by sfackler** [comment]
CI is failing.

File: openssl/src/pkey.rs [hunk 1]
```diff
@@ -627,6 +626,36 @@ impl PKey<Private> {
...
+    /// # Panics
+    ///
+    /// Panics if `passphrase` contains an embedded null.
+    pub fn private_key_to_pkcs8_passphrase(
+        &self,
+        cipher: Cipher,
+        passphrase: &[u8],
+    ) -> Result<Vec<u8>, ErrorStack> {
+        unsafe {
+            ffi::init();
```
Comment (sfackler) [inline openssl/src/pkey.rs:642]: The PKey existing implies this has already happened so you can skip it.

Comment (sven-hm) [reply openssl/src/pkey.rs]: Ok, i removed it and rebased the branch to resolve the new conflicts.

**Comment by sfackler** [comment]
It looks like the definition has changed slightly in OpenSSL 3.0.0.

**Comment by sven-hm** [comment]
Sorry for doing such tiny steps, i missed the `const EVP_PKEY` in version 3.0.0.
I'm not sure what to do about the ["windows-vcpkg" issue in the pipeline](https://github.com/sfackler/rust-openssl/runs/3051920810?check_suite_focus=true), can you provide help there, @sfackler ? Thanks!

**Comment by sfackler** [comment]
Oh right, it's an issue with ctest around functions taking the password callback. You'll want to add the new function here to disable the check on windows: https://github.com/sfackler/rust-openssl/blob/master/systest/build.rs#L106

**Comment by sven-hm** [comment]
Ok, i added the new function to skip the test on windows.
I also rebased and squashed the few fixups.

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**